### PR TITLE
Booking confirm endpoint refactoring

### LINKED ISSUE #2170: Unpaid/Unconfirmed bookings should not block availability
### Issue Summary

Found out that unconfirmed/unpaid bookings are blocking availability when they should only do if they're confirmed/paid.

### Steps to Reproduce

1. Connect a Stripe account
2. Create a priced event type
3. Remove Stripe account (so we can trigger an error)
4. Try to book paid event type
5. You'll get this error `ERROR [api] book:user Creating payment failed`
6. Try to book again, you'll get this error:
    ```sh
    10:34:57.582 timeZoneName DEBUG [api] book:user Booking Pro Example failed
    {
      errorCode: 'BookingUserUnAvailable',
      message: 'Pro Example is unavailable at this time.'
    }
    ```
7. Go to `/bookings` you'll se and unconfirmed/unpaid booking

### LINKED ISSUE #2550: `Unconfirmed` bookings prevent other attendees from booking the same time slot.
### Issue Summary

I created an event-type that requires confirmation. Contrary to expectation, an **unconfirmed** booking is preventing other attendees from booking the same time slot. It should be up to the organizer to pick the attendee they would like to book if there are multiple attendees for the same time slot. **This also leaves room for mischief**: a bad-faith attendee could potentially block an organizer's schedule, creating a situation where the organizer would have to continuously watch for and reject bookings by said attendee.

### Steps to Reproduce

1. Create an event-type that requires confirmation.
2. Click preview to create a booking for this event-type (you're acting as attendee `John` in this step).
3. Do not confirm the booking.
4. Click preview again to create a booking and pick the same time that `John` picked (you're acting as attendee `Bob` in this step).
5. `Bob` will not be able to pick the same time slot that `John` picked *even though* `John's` booking was not confirmed by the organizer.

### Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?
I'm calling this a bug because I don't remember encountering this behaviour in release v1.2. 

### Expected behaviour
The organizer should be able to pick and choose the attendee they would like to book if there are multiple attendees trying to book the same time slot.

### Technical details

No tech specs are relevant here.

@agustif

### LINKED ISSUE #2874: Accepted Opt-in booking still shows as tentative in my Google Calendar
- Someone booked an Opt-in event with me
- I accepted the booking request from the bookings page
- My calendar showed the attendee as confirmed, but I showed up as unconfirmed
- I would expect that accepting the booking in Cal marks me as "Yes" to attend in my Calendar.

https://www.loom.com/share/754335637b734921bab4a0ebd9e1ed75

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Introduces a `module` based project structure (Following admin console findings)
- Adds useful vscode snippets to API handlers
- Fixes unconfirmed bookings should not block availability
- Fixes #2550
- Fixes #2170
- Deprecates `booking.confirmed` over `booking.status: ACCEPTED`
- Deprecates `booking.rejected` over `booking.status: REJECTED`
- Adds visual distinction for opt-in bookings
- Fixes #2874 

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Make an opt-in booking, it should not block availability
- [ ] Organizer should appear as accepted in gCal
- [ ] `yarn test-e2e`

### CODE REVIEW

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -103,11 +102,8 @@ export default class GoogleCalendarService implements Calendar {
           timeZone: calEventRaw.organizer.timeZone,
         },
         attendees: [
-          { ...calEventRaw.organizer, organizer: true },
-          ...calEventRaw.attendees.map((attendee) => ({
-            ...attendee,
-            responseStatus: "accepted",
-          })),
+          { ...calEventRaw.organizer, organizer: true, responseStatus: "accepted" },
+          ...calEventRaw.attendees.map((attendee) => ({ ...attendee, responseStatus: "accepted" })),
```
Comment (zomars) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:106]: fixes #2874

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 2]
```diff
@@ -185,7 +181,7 @@ export default class GoogleCalendarService implements Calendar {
           dateTime: event.endTime,
           timeZone: event.organizer.timeZone,
         },
-        attendees: event.attendees,
+        attendees: [{ ...event.organizer, organizer: true, responseStatus: "accepted" }, ...event.attendees],
```
Comment (zomars) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:184]: Same for rescheduling

**Review by zomars — COMMENTED** [review]
Ready for review

File: apps/web/components/eventtype/EventTypeDescription.tsx [hunk 3]
```diff
@@ -83,6 +82,12 @@ export const EventTypeDescription = ({ eventType, className }: EventTypeDescript
               </IntlProvider>
             </li>
           )}
+          {eventType.requiresConfirmation && (
+            <li className="mr-4 flex items-center whitespace-nowrap">
+              <ClipboardCheckIcon className="mr-1.5 inline h-4 w-4 text-neutral-400" aria-hidden="true" />
+              Opt-in
+            </li>
+          )}
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply apps/web/components/eventtype/EventTypeDescription.tsx:90]: love this

**Review by PeerRich — APPROVED** [review]
great refactor! I love the opt-in badge, too

File: apps/web/components/eventtype/EventTypeDescription.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jaibles) [reply apps/web/components/eventtype/EventTypeDescription.tsx:90]: agreed!

File: apps/web/lib/getBusyTimes.ts [hunk 4]
```diff
@@ -19,24 +19,13 @@ async function getBusyTimes(params: {
...
-              },
-            ],
-          },
-        ],
+        userId,
+        eventTypeId,
+        startTime: { gte: new Date(startTime) },
+        endTime: { lte: new Date(endTime) },
+        status: {
+          in: [BookingStatus.ACCEPTED],
```
Comment (zomars) [inline apps/web/lib/getBusyTimes.ts:27]: This will prevent unconfirmed bookings to block availability as https://github.com/calcom/cal.com/issues/2550

**Pull request merged**